src/main/java/dev/ebullient/convert/tools/dnd5e/JsonSource.java

```diff
@@ -139,7 +138,7 @@ default void appendToText(List<String> text, JsonNode node, String heading) {
     }
 
     default void appendObjectToText(List<String> text, JsonNode node, String heading) {
-        AppendTypeValue type = AppendTypeValue.valueFrom(node, SourceField.type);
+        AppendTypeValue type = SourceField.type.getEnumValueFrom(node, AppendTypeValue.class);
         String source = SourceField.source.getTextOrEmpty(node);
```

Comment:
There is a move here from the use of FieldValues, which was a very clear marker for me.

There are the usual fields, where the enum is the field name/object attribute key, and then there are field values, where the enum defines the allowed values of the field.

I think you're trying to get there through this: 

SourceField.type.getEnumValueFrom(node, AppendTypeValue.class);

but I miss the marker on the type itself (as well as the more limited scope)

Reply:
I've pulled those changes out into a separate commit, and left them out of this PR. I'll play around with it more later probably and try to find a way to nicely replicate the `valueFrom` behavior without losing the inheritance